native/cocos/renderer/gfx-gles2/GLES2Commands.cpp

```diff
@@ -1575,8 +1576,34 @@ static void doCreateFramebufferInstance(GLES2Device *device, GLES2GPUFramebuffer
                                              : gpuFBO->gpuDepthStencilTexture;
         }
 
-        outFBO->framebuffer.initialize(doCreateFramebuffer(device, gpuFBO->gpuColorTextures, colors.data(), colors.size(),
-                                                           depthStencilTexture, resolves, depthStencilResolveTexture, &outFBO->resolveMask));
+        GLES2GPUFramebuffer::GLFramebufferInfo fbInfo;
+        auto wIter = std::min_element(gpuFBO->gpuColorTextures.begin(), gpuFBO->gpuColorTextures.end(), [](const GLES2GPUTexture *lhs, const GLES2GPUTexture *rhs) { return lhs->width < rhs->width; });
+        auto hIter = std::min_element(gpuFBO->gpuColorTextures.begin(), gpuFBO->gpuColorTextures.end(), [](const GLES2GPUTexture *lhs, const GLES2GPUTexture *rhs) { return lhs->height < rhs->height; });
+        fbInfo.width = hIter == gpuFBO->gpuColorTextures.end() ? 0 : (*wIter)->width;
+        fbInfo.height = wIter == gpuFBO->gpuColorTextures.end() ? 0 : (*hIter)->height;
+        if (gpuFBO->gpuDepthStencilTexture) {
+            auto dsWidth = gpuFBO->gpuDepthStencilTexture->width;
+            auto dsHeight = gpuFBO->gpuDepthStencilTexture->height;
+            fbInfo.width = gpuFBO->gpuColorTextures.empty() ? dsWidth : std::min(fbInfo.width, dsWidth);
+            fbInfo.height = gpuFBO->gpuColorTextures.empty() ? dsWidth : std::min(fbInfo.height, dsHeight);
+        }
+
+        fbInfo.glFramebuffer = device->framebufferCacheMap()->getFramebufferByHash(gpuFBO);
```

Comment:
will get by hash cause collision?

Reply:
Hash by `GLES2/3Texture`, in gfx thread there is no `GFXAgent` or `GFXValidator` get involved, it's safe if there's no placement new on `GFXTexture`. To make it more reliable we can hash `glTarget/glTexture/glRenderBuffer` additionally.